deepchecks/vision/vision_data/batch_wrapper.py

```diff
@@ -0,0 +1,179 @@
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Contains code for BatchWrapper."""
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+
+from deepchecks.vision.utils.image_functions import crop_image
+from deepchecks.vision.utils.vision_properties import PropertiesInputType, calc_vision_properties, validate_properties
+from deepchecks.vision.vision_data import VisionData
+from deepchecks.vision.vision_data.utils import BatchOutputFormat, TaskType, sequence_to_numpy
+
+__all__ = ['BatchWrapper']
+
+
+class BatchWrapper:
+    """Represents dataset batch returned by the dataloader during iteration."""
+
+    def __init__(
+            self,
+            batch: BatchOutputFormat,
+            vision_data: VisionData,
+    ):
+        self._vision_data = vision_data
+        self._task_type = vision_data.task_type
+        self._batch = batch
+        self._labels, self._predictions, self._images = None, None, None
+        self._embeddings, self._additional_data, = None, None
+        self._image_identifiers = batch.get('image_identifiers')
+        # if there are no image identifiers, use the number of the image in loading process as identifier
+        if self._image_identifiers is None:
+            images_seen = self._vision_data.number_of_images_cached
+            self._image_identifiers = np.asarray(range(images_seen, images_seen + len(self)), dtype='str')
+
+        self._vision_properties_cache = dict.fromkeys(PropertiesInputType)
+
+    def _get_relevant_data_for_properties(self, input_type: PropertiesInputType,
+                                          classes_to_consider: Optional[List[str]]):
+        result = []
+        if input_type == PropertiesInputType.PARTIAL_IMAGES:
+            for img, bboxes_in_img in zip(self.numpy_images, self.numpy_labels):
+                if bboxes_in_img is None:
+                    continue
+                if classes_to_consider is not None:
+                    bboxes_in_img = bboxes_in_img[bboxes_in_img[:, 0].isin(classes_to_consider)]
+                result = result + [crop_image(img, *bbox[1:]) for bbox in bboxes_in_img]
+        elif input_type == PropertiesInputType.IMAGES:
+            result = self.numpy_images
+        elif input_type == PropertiesInputType.LABELS:
+            result = self.numpy_labels
+        elif input_type == PropertiesInputType.PREDICTIONS:
+            result = self.numpy_predictions
+        return result
+
+    def vision_properties(self, properties_list: List[Dict], input_type: PropertiesInputType,
+                          classes_to_consider: Optional[List[Union[str, int]]] = None):
```

Comment:
so classes_to_consider is just for future stuff?
need to make sure the filtering on the properties will yield same size as the batch with filtering, if not it should happen not here but on the data itself

Reply:
yep, future stuff. point taken - will remove it all together for now